client/state/data-layer/wpcom/gravatar-upload/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	bumpStat,
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'state/analytics/actions';
+
+export function uploadGravatar( { dispatch }, action ) {
+	const { email, file } = action;
+	dispatch( http( {
+		method: 'POST',
+		path: '/gravatar-upload',
+		body: {},
+		apiNamespace: 'wpcom/v2',
+		formData: [
+			[ 'account', email ],
+			[ 'filedata', file ],
+		],
+	}, action ) );
+}
+
+export function announceSuccess( { dispatch }, { file }, next, data ) {
+	// Return a Promise for testing purposes
+	return new Promise( ( resolve, reject ) => {
+		// Sometimes wpcom-proxy-request reports a failed connection as a success, so
+		// we test for the response here. See
+		// https://github.com/Automattic/wp-calypso/pull/15636#issuecomment-312860944
```

Comment:
Would be great if we could just fix `wpcom-proxy-request` before this PR ships so that we don't need this workaround and associated tests.